suplemon/viewer.py

```diff
@@ -50,76 +51,38 @@ def __init__(self, app, window):
         self.x_scroll = 0
         self.cursors = [Cursor()]
 
-        # Lexer for translating tokens to strings
-        self.lexer = None
-        # Built in syntax definition (for commenting etc.)
-        self.syntax = None
-        # Normal Pygments lexer
-        self.pygments_syntax = None
+        # Copy/paste buffer
+        self.buffer = []
+
+        # Runnable methods
+        self.operations = {
+            "arrow_right": self.arrow_right,              # Arrow Right
+            "arrow_left": self.arrow_left,                # Arrow Left
+            "arrow_up": self.arrow_up,                    # Arrow Up
+            "arrow_down": self.arrow_down,                # Arrow Down
+            "jump_left": self.jump_left,                  # Ctrl + Left
+            "jump_right": self.jump_right,                # Ctrl + Right
+            "jump_up": self.jump_up,                      # Ctrl + Up
+            "jump_down": self.jump_down,                  # Ctrl + Down
+            "page_up": self.page_up,                      # Page Up
+            "page_down": self.page_down,                  # Page Down
+            "home": self.home,                            # Home
+            "end": self.end,                              # End
+            "find": self.find_query                       # Ctrl + F
+        }
```

Comment:
I think breaking out operations to their own files is reasonable. However, it's mostly useful when the contents of an operation are overwhelming for a developer. I think they are pretty bite-sized for now so a class works.

My preferred approach for something like that is usually a hybrid; a core class which has lots of small/simple operations with imported external functions for more complex ones:

```python
from suplemon.operations.super_complex import super_complex

class Operations(object):
    def arrow_right:
        # ...

    @classmethod
    def add_operation(cls, name, fn):
      """Helper to attach a new operation to our operations class""
      cls.name = fn

# Add external operations
Operations.add_operation('super_complex', super_complex)
```